Build is failing - SKL 1.3 release

https://github.com/microsoft/hummingbird/actions/runs/5490659732/jobs/10006370962


```
self = <test_sklearn_decomposition.TestSklearnCrossDecomposition testMethod=test_pca_converter_two>

    def test_pca_converter_two(self):
>       self._fit_model_pls_regressor(PLSR(n_components=2))

tests/test_sklearn_decomposition.py:166: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_sklearn_decomposition.py:162: in _fit_model_pls_regressor
    np.testing.assert_allclose(model.predict(X), torch_model.predict(X), rtol=1e-6, atol=2 * 1e-5)
hummingbird/ml/containers/_sklearn_api_containers.py:122: in predict
    return self._run(self._predict, *inputs)
hummingbird/ml/containers/_sklearn_api_containers.py:[70](https://github.com/microsoft/hummingbird/actions/runs/5490659732/jobs/10006370962#step:9:71): in _run
    return function(*inputs)
hummingbird/ml/containers/sklearn/pytorch_containers.py:207: in _predict
    output = self.model.forward(*inputs).cpu().numpy()
hummingbird/ml/_executor.py:115: in forward
    outputs = operator(*(variable_map[input_name] for input_name in operator.inputs))
/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/torch/nn/modules/module.py:1501: in _call_impl
    return forward_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = CrossDecomposition()
x = tensor([[-1.3056, -0.[74](https://github.com/microsoft/hummingbird/actions/runs/5490659732/jobs/10006370962#step:9:75)06, -0.4392],
        [-0.2611, -0.7406, -1.0247],
        [ 0.7833,  0.1058,  0.1464],
        [ 0.7833,  1.3[75](https://github.com/microsoft/hummingbird/actions/runs/5490659732/jobs/10006370962#step:9:76)4,  1.3175]])

    def forward(self, x):
        x -= self.x_mean
        x /= self.x_std
>       y_pred = torch.mm(x, self.coefficients).float()
E       RuntimeError: mat1 and mat2 shapes cannot be multiplied (4x3 and 2x3)

hummingbird/ml/operator_converters/_decomposition_implementations.py:100: RuntimeError
________ TestSklearnCrossDecomposition.test_pca_converter_two_no_scale _________

self = <test_sklearn_decomposition.TestSklearnCrossDecomposition testMethod=test_pca_converter_two_no_scale>

    def test_pca_converter_two_no_scale(self):
>       self._fit_model_pls_regressor(PLSR(n_components=3, scale=False))

tests/test_sklearn_decomposition.py:170: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_sklearn_decomposition.py:162: in _fit_model_pls_regressor
    np.testing.assert_allclose(model.predict(X), torch_model.predict(X), rtol=1e-6, atol=2 * 1e-5)
hummingbird/ml/containers/_sklearn_api_containers.py:122: in predict
    return self._run(self._predict, *inputs)
hummingbird/ml/containers/_sklearn_api_containers.py:70: in _run
    return function(*inputs)
hummingbird/ml/containers/sklearn/pytorch_containers.py:207: in _predict
    output = self.model.forward(*inputs).cpu().numpy()
hummingbird/ml/_executor.py:115: in forward
    outputs = operator(*(variable_map[input_name] for input_name in operator.inputs))
/opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/torch/nn/modules/module.py:1501: in _call_impl
    return forward_call(*args, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = CrossDecomposition()
x = tensor([[-1.2500, -1.7500, -0.7500],
        [-0.2500, -1.7500, -1.7500],
        [ 0.7500,  0.2500,  0.2500],
        [ 0.7500,  3.2500,  2.2500]])

    def forward(self, x):
        x -= self.x_mean
        x /= self.x_std
>       y_pred = torch.mm(x, self.coefficients).float()
E       RuntimeError: mat1 and mat2 shapes cannot be multiplied (4x3 and 2x3)
```

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -123,13 +123,14 @@ jobs:
       working-directory: ../tvm/build
       run: |
         make -j3
-        python -m pip install -U wheel packaging
-        python setup.py bdist_wheel
     - name: Install python TVM if Mac
       if: ${{  startsWith(matrix.os, 'macos') }}
       working-directory: ../tvm/python
       run: |
+        python -m pip install -U wheel packaging
+        python setup.py bdist_wheel
         python -m pip install dist/tvm-*.whl
+
     # We don't run pytest for Linux py3.8 since we do coverage for that case.
     - name: Test with pytest
       if: ${{ matrix.python-version != '3.8' || startsWith(matrix.os, 'ubuntu') != true }}

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -49,7 +49,7 @@ jobs:
       run: |
         python -m pip install -e .[tests] -f https://download.pytorch.org/whl/torch_stable.html
     - name: Test with older SKLearn on Linux with py3.9 to check backward compatibility
-      if: ${{ matrix.python-version == '3.9' || startsWith(matrix.os, 'ubuntu') == true }}
+      if: ${{ matrix.python-version == '3.9' && startsWith(matrix.os, 'ubuntu') == true }}
       run: python -m pip install scikit-learn==1.2.1
     # Compile (but not push) documentation only for one of the runs (Linux py3.8).
     - name: Generate Documentation

--- a/.github/workflows/pythonapp.yml
+++ b/.github/workflows/pythonapp.yml
@@ -11,7 +11,7 @@ on:
   pull_request:
     branches:
       - main
-      
+
   schedule:
     - cron: "15 21 * * FRI"
 
@@ -48,6 +48,9 @@ jobs:
     - name: Install basic dependencies
       run: |
         python -m pip install -e .[tests] -f https://download.pytorch.org/whl/torch_stable.html
+    - name: Test with older SKLearn on Linux with py3.9 to check backward compatibility
+      if: ${{ matrix.python-version == '3.9' || startsWith(matrix.os, 'ubuntu') == true }}
+      run: python -m pip install scikit-learn==1.2.1
     # Compile (but not push) documentation only for one of the runs (Linux py3.8).
     - name: Generate Documentation
       if: ${{ matrix.python-version == '3.8' && startsWith(matrix.os, 'ubuntu') }}

--- a/hummingbird/ml/operator_converters/sklearn/decomposition.py
+++ b/hummingbird/ml/operator_converters/sklearn/decomposition.py
@@ -135,7 +135,11 @@ def convert_sklearn_pls_regression(operator, device, extra_config):
     """
     assert operator is not None, "Cannot convert None operator"
 
-    coefficients = operator.raw_operator.coef_
+    # Check if operator.raw_operator._coef_ does not exist, which means to transpose coefficients (#26016)
+    if hasattr(operator.raw_operator, "_coef_"):  # SKL<1.3
+        coefficients = operator.raw_operator.coef_
+    else:  # SKL>=1.3
+        coefficients = operator.raw_operator.coef_.T
     x_mean = operator.raw_operator._x_mean
     x_std = operator.raw_operator._x_std
     y_mean = operator.raw_operator._y_mean

